extensions/NexusKitten/moremotion.js

```diff
@@ -0,0 +1,294 @@
+            text: 'distance from x: [X] y: [Y]',
+            arguments: {
+              X: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: '0'
+              },
+              Y: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: '0'
+              }
+            }
+          },
+          {
+            filter: [Scratch.TargetType.SPRITE],
+            opcode: 'spritewh',
+            blockType: Scratch.BlockType.REPORTER,
+            text: 'sprite [WHAT]',
+            disableMonitor: true,
+            arguments: {
+              WHAT: {
+                type: Scratch.ArgumentType.STRING,
+                menu: 'WHAT'
+              }
+            }
+          },
+          '---',
+          {
+            filter: [Scratch.TargetType.SPRITE],
+            opcode: 'touchingxy',
+            blockType: Scratch.BlockType.BOOLEAN,
+            text: 'touching x: [X] y: [Y]?',
+            arguments: {
+              X: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: '0'
+              },
+              Y: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: '0'
+              }
+            }
+          },
+          {
+            filter: [Scratch.TargetType.SPRITE],
+            opcode: 'inbetween',
+            blockType: Scratch.BlockType.BOOLEAN,
+            text: 'in-between x1: [X1] y1: [Y1] x2: [X2] y2: [Y2]?',
+            arguments: {
+              X1: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: '-100'
+              },
+              Y1: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: '-100'
+              },
+              X2: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: '100'
+              },
+              Y2: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: '100'
+              }
+            }
+          },
+        ],
+        menus: {
+          WHAT: {
+            acceptreporters: false,
+            items: ['width', 'height']
+          }
+        }
+      };
+    }
+
+    changexy(args, util) {
+      const x = Scratch.Cast.toNumber(args.X);
+      const y = Scratch.Cast.toNumber(args.Y);
+      util.target.setXY(util.target.x + x, util.target.y + y);
+    }
+
+    // LORAX APPROVED
+    pointto(args, util) {
+      const x = Scratch.Cast.toNumber(args.X);
+      const y = Scratch.Cast.toNumber(args.Y);
+      if (util.target.y > y) {
+        util.target.setDirection(((180 / Math.PI) * Math.atan((x - util.target.x) / (y - util.target.y))) + 180);
+      } else {
+        util.target.setDirection(((180 / Math.PI) * Math.atan((x - util.target.x) / (y - util.target.y))));
+      }
+    }
+
+    rotationStyle(args, util) {
+    return util.target.rotationStyle;
+    }
+
+    fence(util) {

```

Comment:
- first argument is always args (an empty object in this case). second argument is always util.
 - you should make the fencing match Scratch. here's a hint: https://github.com/TurboWarp/scratch-vm/blob/9329427fb57de83cf5d8c2774b55013c60f39ed4/src/sprites/rendered-target.js#L269

Reply:
All issues you presented should be resolved in the new commit.